contracts/Staking.sol

```diff
@@ -780,25 +783,26 @@ contract Staking is Pauser, Whitelist {
     /**
      * Create validator set for an epoch
      */
-    function createEpochValidators(bytes memory _epochHeaderBytes) public {
-        // TODO:
-        // epochHeader contains
-        // - commit from epoch (epochIdx - 1)'s validators
-        // - new validators for epoch epochIdx
-        // Verify commit using current validators
-        // call _createEpochValidators()
-    }
+    function createEpochValidators(bytes memory _epochHeaderBytes, bytes memory commitBytes) public {
+        //1. verify epoch header
+        (bool succeed, uint256 height, bytes32 headerHash) = BlockDecoder.verifyHeader(
+            _epochHeaderBytes,
+            commitBytes,
+            currentEpochIdx,
+            currentVotingPowers
+        );
 
-    function getEpochValidators()
-        public
-        view
-        returns (
-            uint256,
-            address[] memory,
-            uint256[] memory
-        )
-    {
-        return (epochIdx + 1, proposedValidators, proposedVotingPowers);
+        require(succeed, "invalid header");
+
+        require(curHeight < height, "too small height");
```

Comment:
Per our yesterday's discussion, a couple of checks may be needed here:
1, check if the height is for an epoch header?
2, check if the epoch equals current epoch + 1?
3, if nextvalidators() is zero, use current validators and powers (i.e., unchanged)?